### PR TITLE
Add buttons

### PR DESCRIPTION
**Pull request by bazelajoanna**
Create Button component that handles all the management around disabled state, primary and non primary styling, download icon rendering and children rendering.

### CODE REVIEW

**Comment by bazelajoanna** [comment]
@evulins - ready to be reviewed :)

File: src/assets/styles/colors.js [hunk 1]
```diff
@@ -32,7 +34,7 @@ const mainHeader = "#161C56";
 
 const primarySubtitle = "#124CAB";
 
-const seconderySubtitle = "#F88B0C";
+const secondarySubtitle = "#F88B0C";
```
Comment (evulins) [inline src/assets/styles/colors.js:37]: 👍

File: src/components/Button/interface.tsx [hunk 2]
```diff
@@ -0,0 +1,5 @@
+export interface ButtonProps {
```
Comment (evulins) [inline src/components/Button/interface.tsx]: Could we use in this case` type PropsOptional`?

File: src/components/Button/styledComponents.tsx [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+import {
+  activeButton,
+  hoverButton,
+  disabledButton,
+  hoverSecondaryButton,
+} from "../../assets/styles/colors";
+import { ReactComponent as Download } from "../../assets/img/icons/download.svg";
+import { minDevice } from "../../assets/styles/breakpoints";
+
+interface ButtonStyleProps {
```
Comment (evulins) [inline src/components/Button/styledComponents.tsx]: How about moving it to the interface file? Is this making any sense?

File: src/components/Button/styledComponents.tsx [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+    background-color: ${disabledButton};
+    box-shadow: none;
+  }
+
+  @media ${minDevice.desktopL} {
+    padding: 16px 48px;
+    font-size: 1rem;
+  }
+
+  ${(props) =>
```
Comment (evulins) [inline src/components/Button/styledComponents.tsx:47]: 🥇

File: src/components/ProblemIndex/index.tsx [hunk 5]
```diff
@@ -0,0 +1,29 @@
+import { FC } from "react";
+
+import ResultCountLabel from "../ResultCountLabel";
+import Tag from "../Tag";
+import Button from "../Button";
+
+const ProblemIndex: FC = () => {
```
Comment (evulins) [inline src/components/ProblemIndex/index.tsx:7]: Great idea!

File: src/components/Wrapper/interface.ts [hunk 6]
```diff
@@ -0,0 +1,12 @@
+interface WraperProps {
```
Comment (evulins) [inline src/components/Wrapper/interface.ts:1]: 👍

**Review by evulins — CHANGES_REQUESTED** [review]
Great job! I have only few questions :)

File: src/components/Button/interface.tsx (hunk 2 above)
Comment (bazelajoanna) [reply src/components/Button/interface.tsx]: We don't need any of these props to be required elsewhere and having them optional is the only version of interface we need in our app - that's why we don't need to create another type with using PropsOptional.

File: src/components/Button/styledComponents.tsx (hunk 3 above)
Comment (bazelajoanna) [reply src/components/Button/styledComponents.tsx]: Yes, good point :) I've moved it there.

**Review by evulins — APPROVED** [review]

**Pull request merged**